### PR TITLE
Extra key for HTTP error responses

### LINKED ISSUE #4284: Provide extra info along "Invalid tx" error
Related issue https://github.com/aeternity/aeternity/issues/4283.

Currently, if for some reason I posted (to `/transactions`) an invalid transaction I'm getting `{"reason":"Invalid tx"}`. Various mistakes end up with this message, like:
- invalid signature,
- low nonce,
- low ttl,
- low fee.

Would be good if the node provide some key to indicate exactly the first check failed, like in `debug/check-tx/pool`. This would allow tools (like js-sdk) to provide developers with more friendly error messages, to simplify debugging in some cases.

As far as these checks are already done by the node while accepting transaction, it shouldn't require extra computations to provide it in response.

### PR DESCRIPTION
**Pull request by ThomasArts**
Fixes #4284 

This PR is supported by Æternity Foundation

### CODE REVIEW

**Review by hanssv — APPROVED** [review]

File: apps/aehttp/priv/swagger.yaml [hunk 1]
```diff
@@ -3281,6 +3281,8 @@ definitions:
     properties:
       reason:
         type: string
+      error_code:
```
Comment (davidyuk) [inline apps/aehttp/priv/swagger.yaml:3284]: can it be just `code`? As far as it is already in context of `Error`
seems I have troubles with snake keys in node's exceptions

**Review by davidyuk — APPROVED** [review]
It is fine, the only proposal is to rename `error_code` to `code`

File: apps/aehttp/priv/oas3.yaml [hunk 2]
```diff
@@ -3378,6 +3378,8 @@ components:
       properties:
         reason:
           type: string
+        error_code:
+          type: string
       required:
         - reason
```
Comment (davidyuk) [inline apps/aehttp/priv/oas3.yaml:3384]: `error_code` was added in other places, but there are still cases when it is missed? so it can't be required?

Comment (hanssv) [reply apps/aehttp/priv/oas3.yaml:3384]: It was only added where there is anything interesting to report I think.

File: apps/aehttp/priv/swagger.yaml (hunk 1 above)
Comment (ThomasArts) [reply apps/aehttp/priv/swagger.yaml:3284]: This can be confusing with contract code, so rather not.

File: apps/aehttp/priv/oas3.yaml (hunk 2 above)
Comment (ThomasArts) [reply apps/aehttp/priv/oas3.yaml:3384]: And it is not required. You can get responses without any `error_code`. It's just a bit of extra information the SDK can look at when deciding what to do.

**Review by dincho — APPROVED** [review]

**Pull request merged**